e2e/app/component/share-modal.ts

```diff
@@ -0,0 +1,107 @@
+import Container from 'app/container';
+import {ElementHandle, Page} from 'puppeteer';
+import {WorkspaceAccessLevel} from 'app/page-identifiers';
+import {ElementType} from 'app/xpath-options';
+import Button from 'app/element/button';
+import Textbox from 'app/element/textbox';
+import ClrIcon from 'app/element/clr-icon-link';
+
+
+export enum ButtonLabel {
+  Cancel = 'Cancel',
+  Save = 'Save'
+}
+
+const Selector = {
+  defaultDialog: '//*[@role="dialog"]',
+}
+
+export default class ShareModal extends Container {
```

Comment:
Can you change to `ShareModal extends Dialog` so common methods are not duplicated?

Reply:
Done